sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/QuickPulseDataSender.java

```diff
@@ -79,13 +79,15 @@ public void run() {
             dataPointList.add(point);
             Date currentDate = new Date();
             long transmissionTimeInTicks = currentDate.getTime() * 10000 + TICKS_AT_EPOCH;
+            String etag = configuration.get().getETag();
+
+            logger.verbose("Attempting to send data points to quickpulse with etag {}: {}", etag,
+                printListOfMonitoringPoints(dataPointList));
+
```

Comment:
`printListOfMonitoringPoints` will be executed even if the logging level is not verbose, inducing extra processing with a `StringBuilder`.